### PR TITLE
Pass http proxy as env to command (fix #136, #117)

### LINKED ISSUE #136: mac login error
login: pass: - Signing in leetcode.com
[ERROR] Error: tunneling socket could not be established, cause=connect EHOSTUNREACH 0.0.4.63:80 - Local (10.0.0.11:59146)

### PR DESCRIPTION
**Pull request by edvardchen**
We use leetcode-cli and leetcode-cli uses request to call LeetCode web API.
It's lucky that request respects the http proxy env variables(http_proxy, https_proxy),
so all we need to do is passing proxy as env variable http_proxy to every command.

TODO: do we need to set https_proxy for https request?

### CODE REVIEW

**Comment by jdneo** [comment]
Thank you @edvardchen for contribution.

So Is this PR aimed to fix #117

**Comment by edvardchen** [comment]
Yes.

**Comment by jdneo** [comment]
@edvardchen Awesome, I'll take a look later, thanks!

**Comment by jdneo** [comment]
Hi @edvardchen

What will happen if we don't set https_proxy for https request

**Comment by edvardchen** [comment]
request will use http_proxy as fallback

**Comment by edvardchen** [comment]
Because vscode user setting doesn't have a https.proxy option, so we should not worry about it.

File: src/utils/workspaceUtils.ts [hunk 1]
```diff
@@ -44,3 +44,18 @@ export async function getActiveFilePath(uri?: vscode.Uri): Promise<string | unde
 export function getWorkspaceConfiguration(): vscode.WorkspaceConfiguration {
     return vscode.workspace.getConfiguration("leetcode");
 }
+
+function getHttpAgent(): string | undefined {
+    return vscode.workspace.getConfiguration("http").get<string>("proxy");
+}
+
+// clone process.env and add http proxy
+export function createEnvOption(): {} {
```
Comment (jdneo) [inline src/utils/workspaceUtils.ts]: Looks like cpUtils.ts is a better place to put this method?

Comment (edvardchen) [reply src/utils/workspaceUtils.ts]: Sure. I'll move it there.

**Comment by jdneo** [comment]
@edvardchen Thank you for your contribution!

**Pull request merged**